### PR TITLE
Add settingEngine hooks to modify ICE timeouts

### PR DESCRIPTION
**Pull request by wdouglass**
Relates to #324

### CODE REVIEW

File: pkg/ice/agent.go [hunk 1]
```diff
@@ -126,6 +137,19 @@ func NewAgent(config *AgentConfig) (*Agent, error) {
 		portmax:     config.PortMax,
 	}
 
+	// connectionTimeout used to declare a connection dead
+	if config.ConnectionTimeout == 0 {
```
Comment (Sean-Der) [inline pkg/ice/agent.go]: I can't think of a case, but could 0 ever be an actual value someone wants? Maybe disabling timeout all together? (Seems unlikely though)

File: rtcpeerconnection.go [hunk 2]
```diff
@@ -158,7 +158,9 @@ func New(configuration RTCConfiguration) (*RTCPeerConnection, error) {
 		}
 	}
 
-	pc.networkManager, err = network.NewManager(urls, pc.generateChannel, pc.iceStateChange, defaultSettingEngine.EphemeralUDP.PortMin, defaultSettingEngine.EphemeralUDP.PortMax)
+	pc.networkManager, err = network.NewManager(urls, pc.generateChannel, pc.iceStateChange,
```
Comment (Sean-Der) [inline rtcpeerconnection.go]: Can we just pass the `ice.AgentConfig` directly here? Having lots of parameters is hard to read (and easy to get them out of order by accident)

File: pkg/ice/agent.go (hunk 1 above)
Comment (wdouglass) [reply pkg/ice/agent.go]: That's a good idea, i'll look into that

File: rtcpeerconnection.go (hunk 2 above)
Comment (wdouglass) [reply rtcpeerconnection.go]: It's public, I don't see why not

File: pkg/ice/agent.go (hunk 1 above)
Comment (wdouglass) [reply pkg/ice/agent.go]: Does it make more sense to create a new null.Duration object, or to just use a pointer and create a copy when the user calls SetConnectionTimeout?

The third option, which i don't like but am including here for completeness, is to use null.Int32 and cast back and forth to time.Duration (it's a typedef underneath it all anyway, but I think this is probably less clear...)

Comment (wdouglass) [reply pkg/ice/agent.go]: I've gone with the second option to
a) keep the AgentConfig struct a bit smaller
b) avoid creating a new null.Duration type

Changing over to use pkg/null is easy if you'd like; this just fits my instinct better.

**Pull request merged**